app/models/user.rb

```diff
@@ -56,10 +56,10 @@ class User < ApplicationRecord
 
   scope :ordered, -> { order(username: :asc) }
 
-  def self.from_omniauth(auth, current_user)
-    authorization = Authorization.where(provider: auth.provider, uid: auth.uid.to_s).first_or_create
+  def self.from_omniauth(auth)
+    authorization = Authorization.where(provider: auth[:provider], uid: auth[:uid].to_s).first_or_create
     if authorization.user.blank?
```

Comment:
what do you think about this?
```
return authorization.user if authorization.user
User.find_or_create_by(email: auth.info.email) do |user|
  user.password = Devise.friendly_token[0, 20]
  user.fetch_details(auth)
  user.authorizations << authorization
end
```

Reply:
it looks shorter, and almost without conditions :)

Reply:
return authorization.user if authorization.user
    user = User.find_or_create_by(email: auth[:info][:email]) do |u|
      u.password = Devise.friendly_token[0, 20]
      u.fetch_details(auth)
    end
    user.authorizations << authorization
    user